### PR TITLE
Caps on acronyms

### PR DESCRIPTION
**Pull request by mcardosos**
Fixes https://github.com/Azure/autorest/issues/2224
PTAL
@marstr @jhendrixMSFT

### CODE REVIEW

**Comment by mcardosos** [comment]
Generated code example
``` go
// Create adds an API to the specified product.
//
// resourceGroupName is the name of the resource group. serviceName is the name
// of the API Management service. productID is product identifier. Must be
// unique in the current API Management service instance. APIID is API
// identifier. Must be unique in the current API Management service instance.
func (client ProductApisClient) Create(resourceGroupName string, serviceName string, productID string, APIID string) (result autorest.Response, err error) {
```

File: src/generator/AutoRest.Go/Extensions.cs [hunk 1]
```diff
@@ -53,10 +53,32 @@ public static string ToSentence(this string value)
...
                 value = value.Trim();
+                if (value.StartsWithAcronym())
+                {
+                    return value;
+                }                
                 return value.First().ToString().ToLowerInvariant() + (value.Length > 1 ? value.Substring(1) : "");
             }
         }
 
+        public static bool StartsWithAcronym(this string value)
```
Comment (jhendrixMSFT) [inline src/generator/AutoRest.Go/Extensions.cs]: Can you please add a comment explaining the algorithm this function uses?  It looks like it iterates over the string and if it's all caps then it returns true (seems reasonable however it does depend on the swagger properly casing them).

Comment (mcardosos) [inline src/generator/AutoRest.Go/Extensions.cs]: Yes, it depends on the correct casing in swagger.

Comment (mcardosos) [inline src/generator/AutoRest.Go/Extensions.cs]: Fixed

File: src/generator/AutoRest.Go/Extensions.cs [hunk 2]
```diff
@@ -53,11 +53,39 @@ public static string ToSentence(this string value)
...
+                {
+                    if (char.IsLower(words.First()[i]))
+                    {
+                        return false;
+                    }
+                }
+                return true;
+            }
+            return false;
+        }
```
Comment (marstr) [inline src/generator/AutoRest.Go/Extensions.cs]: I think this method could be tidied up!

``` C#
public static bool StartsWithAcronym(this string value)
{
    value = value.Trim();
    var firstWord = value.Split(' ', '-','_').First();
    var lowercaseLetters = firstWord.Where(c => char.IsLower(c))
    return firstWord.Length > 1 && lowercaseLetters.Any()
}
```

Comment (mcardosos) [inline src/generator/AutoRest.Go/Extensions.cs]: Fixed (even fancier!)

Comment (marstr) [inline src/generator/AutoRest.Go/Extensions.cs]: Haha, so it is!

**Pull request merged**